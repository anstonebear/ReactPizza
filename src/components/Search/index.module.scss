.root {
	position: relative;
	left: -120px;
}

.input {
	border: 1px solid rgba(0, 0, 0, 0.1);
	padding: 8px 15px;
	width: 400px;
	border-radius: 10px;
	font-size: 16px;
}

.clear {
	width: 18px;
	height: 18px;
	opacity: 0.3;
	position: absolute;
	right: 15px;
	top: 10px;
	cursor: pointer;

	&:hover {
		opacity: 0.8;
	}
}
