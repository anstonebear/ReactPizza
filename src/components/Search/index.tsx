import React, { useCallback, useRef, useState } from 'react';

import styles from './index.module.scss';
import debounce from 'lodash.debounce';

interface ISearchProps {
	setSearchValue: (value: string) => void;
}

export const Search: React.FC<ISearchProps> = ({ setSearchValue }) => {
	const [value, setValue] = useState('');

	const updateSearchValue = useCallback(
		debounce((str: string) => {
			setSearchValue(str);
		}, 1000),
		[setSearchValue]
	);

	const onChangeInput = (e: any) => {
		setValue(e.target.value);
		updateSearchValue(e.target.value);
	};

	return (
		<div className={styles.root}>
			<input
				value={value}
				onChange={onChangeInput}
				placeholder='Поиск пиццы...'
				type='search'
				className={styles.input}
			/>
		</div>
	);
};

export default Search;
