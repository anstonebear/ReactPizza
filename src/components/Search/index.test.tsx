import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';

import Search from './index';

describe('Search', () => {
	test('Search', () => {
		const setSearchValueMock = jest.fn();
		render(
			<BrowserRouter>
				<Search setSearchValue={setSearchValueMock} />
			</BrowserRouter>
		);
		const input = screen.getByPlaceholderText(/Поиск пиццы.../i);
		userEvent.click(input);
		expect(input).toBeInTheDocument();
		expect(input).toMatchSnapshot();
		console.log(input);
	});

	test('onChange', () => {
		const input = document.createElement('input');
		const onChangeInput = jest.fn();

		// Call the onChangeInput function with a mock event
		input.onchange = onChangeInput;
		input.dispatchEvent(new Event('change'));

		expect(onChangeInput).toHaveBeenCalled();
	});
});
