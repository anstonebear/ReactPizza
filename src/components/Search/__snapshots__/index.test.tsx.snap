// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Search 1`] = `
<input
  class="input"
  placeholder="Поиск пиццы..."
  type="search"
  value=""
/>
`;
